File: Data Structures & Algorithms/jump-game-ii/submission-0.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int right = 0;
        int left = 0;
        int jump = 0;
        while(right<nums.size()-1){
            int farthest = 0;
            for(int i=left; i<=right; i++){
                farthest = max(farthest, i+nums[i]);
            }
            left = right+1;
            right = farthest;
            jump++;
        }
        return jump;
    }
};

File: Data Structures & Algorithms/longest-strictly-increasing-or-strictly-decreasing-subarray/submission-0.java
class Solution {
    public int longestMonotonicSubarray(int[] nums) {
        
        int count = 1;
        int ans = 0;
        for(int i = 1; i<nums.length; i++){
            if(nums[i]>nums[i-1]){
                count++;
            }
            else{
                ans = Math.max(ans, count);
                count = 1;
            }
        }
        ans = Math.max(count, ans);
        count = 1;
        for(int i = 1; i<nums.length; i++){
            if(nums[i]<nums[i-1]){
                count++;
            }
            else{
                ans = Math.max(ans, count);
                count = 1;
            }
        }
        ans = Math.max(count, ans);
        return ans;
    }
}